elyra/pipeline/pipeline_definition.py

```diff
@@ -432,6 +442,46 @@ def validate(self) -> list:
 
         return validation_issues
 
+    def propagate_pipeline_default_properties(self):
+        """
+        For any default pipeline properties set (e.g. runtime image, volume), propagate
+        the values to any nodes that do not set their own value for that property.
+        """
+        pipeline_default_properties = self.primary_pipeline.get_property(pipeline_constants.PIPELINE_DEFAULTS_NAME, {})
+        for pipeline_default_prop, pipeline_default_value in pipeline_default_properties.items():
+            if not pipeline_default_value:
+                continue
+
+            for pipeline in self.pipelines:
+                for node in pipeline.nodes:
+                    if Operation.is_generic_operation(node.op):
+                        node_value = node.get_component_parameter(pipeline_default_prop)
+                        if not node_value:
+                            node.set_component_parameter(pipeline_default_prop, pipeline_default_value)
+                        else:
+                            if pipeline_default_prop == pipeline_constants.ENV_VARIABLES:
+                                # Transform both into dicts
+                                pipeline_default_env_dict = self.envs_to_dict(env_list=pipeline_default_value)
+                                node_env_dict = self.envs_to_dict(env_list=node_value)
+                                merged_env_list = self.env_dict_to_list({**pipeline_default_env_dict, **node_env_dict})
+                                node.set_component_parameter(pipeline_default_prop, merged_env_list)
+
+    def envs_to_dict(self, env_list: List) -> Dict[str, str]:
+        envs = {}
+        for nv in env_list:
+            if nv:
+                nv_pair = nv.split("=", 1)
+                if len(nv_pair) == 2 and nv_pair[0].strip():
+                    if len(nv_pair[1]) > 0:
+                        envs[nv_pair[0]] = nv_pair[1]
+        return envs
+
+    def env_dict_to_list(self, env_dict: Dict) -> List[str]:
+        envs = []
+        for env, env_value in env_dict.items():
+            envs.append(f"{env}={env_value}")
```

Comment:
Heads up.  If the pipeline default envs are not fully expressed (i.e., they don't all contain non-None values, then this can result in an entry of `foo=None` and won't be discarded in [`GenericOperation.env_vars_as_dict()`](https://github.com/elyra-ai/elyra/blob/6da8f08e31d5bd59a6f7c431158036092399e80f/elyra/pipeline/pipeline.py#L328).

Reply:
Sorry, I didn't see the conversion of the pipeline defaults to discard missing values.  Please disregard my previous comment.